### DISCUSSION TITLE
Requesting support for Cython > 3

### DISCUSSION THREAD

**Discussion by ppfeister** [explosion/spaCy#13601] [New Features & Project Ideas]
Back in spaCy v3.0.7, an upper limit was added to the dependency on Cython. This release set the max version for Cython at 3. Cython of versions below 3 are now no longer available via certain repositories, leading to spaCy's inability to be packaged in those environments without bucking the requirements at risk of breaking certain functionality.

Requesting support for Cython > 3.

For those that remember why this cap was put in place, it would be interesting to know if the blocker is still present.

**Comment by kinghuang** [comment]
Looks like related to #8261, and restricted in #8337.

**Reply by andrey-khropov** [reply]
> Looks like related to https://github.com/explosion/spaCy/pull/8261

Yes, errors reported in this PR are still present if you try to build spaCy from the latest release revision. 
Judging by the error message I guess it is related to the switch to [PEP 563](https://peps.python.org/pep-0563/) - style annotations as strings in Cython 3: https://github.com/cython/cython/issues/2863 which was problematic for Pydantic (that reports errors here): https://github.com/pydantic/pydantic/issues/2678. This has been a big enough deal that it even caused abandonment of the plan to enable [PEP 563](https://peps.python.org/pep-0563/) - style annotations as strings by default in CPython.
